src/main/java/org/matsim/run/rebalancing/WaitingPointsBasedRebalancingStrategy.java

```diff
@@ -0,0 +1,117 @@
+package org.matsim.run.rebalancing;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang.mutable.MutableInt;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.contrib.drt.optimizer.rebalancing.RebalancingParams;
+import org.matsim.contrib.drt.optimizer.rebalancing.RebalancingStrategy;
+import org.matsim.contrib.dvrp.fleet.DvrpVehicle;
+import org.matsim.contrib.dvrp.fleet.Fleet;
+import org.matsim.contrib.dvrp.schedule.Schedule;
+import org.matsim.contrib.dvrp.schedule.Schedules;
+import org.matsim.contrib.dvrp.schedule.StayTask;
+import org.matsim.core.utils.geometry.CoordUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * @author Chengqi Lu
+ */
+public class WaitingPointsBasedRebalancingStrategy implements RebalancingStrategy {
```

Comment:
I think this class should not be public

Reply:
Why not? Maybe not in the kelheim scenario. But probably in the matsim-lib after it is moved there, just like the other rebalancing strategies. 😀

Reply:
Because the `WaitingPointsBasedRebalancingModule` is what is the outward-facing API. According to KN, we want as less public footprint as possible, in order to be able to change internal API.